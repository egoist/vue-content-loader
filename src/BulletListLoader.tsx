import { defineComponent } from 'vue'
import ContentLoader from './ContentLoader'

const BulletListLoader = defineComponent((props, { attrs }) => {
  return () => (
    <ContentLoader {...attrs}>
      <circle cx="10" cy="20" r="8" />
      <rect x="25" y="15" rx="5" ry="5" width="220" height="10" />
      <circle cx="10" cy="50" r="8" />
      <rect x="25" y="45" rx="5" ry="5" width="220" height="10" />
      <circle cx="10" cy="80" r="8" />
      <rect x="25" y="75" rx="5" ry="5" width="220" height="10" />
      <circle cx="10" cy="110" r="8" />
      <rect x="25" y="105" rx="5" ry="5" width="220" height="10" />
    </ContentLoader>
  )
})

export default BulletListLoader
