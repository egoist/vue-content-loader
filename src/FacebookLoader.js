import ContentLoader from './ContentLoader'

const FacebookLoader = (props, { attrs }) => {
  return (
    <ContentLoader {...attrs}>
      <rect x="70" y="15" rx="4" ry="4" width="117" height="6.4" />
      <rect x="70" y="35" rx="3" ry="3" width="85" height="6.4" />
      <rect x="0" y="80" rx="3" ry="3" width="350" height="6.4" />
      <rect x="0" y="100" rx="3" ry="3" width="380" height="6.4" />
      <rect x="0" y="120" rx="3" ry="3" width="201" height="6.4" />
      <circle cx="30" cy="30" r="30" />
    </ContentLoader>
  )
}

export default FacebookLoader
