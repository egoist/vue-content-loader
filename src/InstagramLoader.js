import ContentLoader from './ContentLoader'

export default {
  name: 'InstagramLoader',
  functional: true,
  render(h, { data }) {
    return (
      <ContentLoader {...data} height={480}>
        <circle cx="20" cy="32" r="10.5" />

        <rect x="40" y="27" rx="2" ry="2" width="140" height="10" />
        <rect x="0" y="60" rx="2" ry="2" width="400" height="400" />
        <rect x="0" y="470" rx="2" ry="2" width="140" height="10" />
        <rect x="0" y="485" rx="2" ry="2" width="250" height="10" />
        <rect x="0" y="500" rx="2" ry="2" width="110" height="10" />
        <rect x="0" y="515" rx="2" ry="2" width="180" height="10" />
        <rect x="0" y="530" rx="2" ry="2" width="170" height="10" />
      </ContentLoader>
    )
  }
}
