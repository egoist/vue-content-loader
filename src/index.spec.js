import { mount } from '@vue/test-utils'

import BulletListLoader from './BulletListLoader'
import CodeLoader from './CodeLoader'
import FacebookLoader from './FacebookLoader'
import ListLoader from './ListLoader'
import InstagramLoader from './InstagramLoader'

describe.each([
  ['BulletListLoader', BulletListLoader],
  ['CodeLoader', CodeLoader],
  ['FacebookLoader', FacebookLoader],
  ['ListLoader', ListLoader],
  ['InstagramLoader', InstagramLoader]
])('%s', (name, component) => {
  it('renders its shapes', () => {
    const wrapper = mount(component)

    expect(wrapper.findAll('clipPath > *').map(c => c.html())).toMatchSnapshot()
  })

  it('forwards attributes to ContentLoader', () => {
    const wrapper = mount(component, {
      props: {
        id: 'loader',
        class: 'loader'
      }
    })

    expect(wrapper.attributes()).toMatchObject({
      id: 'loader',
      class: 'loader'
    })
  })
})
