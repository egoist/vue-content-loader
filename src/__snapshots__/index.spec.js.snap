// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletListLoader renders its shapes 1`] = `
Array [
  "<circle cx=\\"10\\" cy=\\"20\\" r=\\"8\\"></circle>",
  "<rect x=\\"25\\" y=\\"15\\" rx=\\"5\\" ry=\\"5\\" width=\\"220\\" height=\\"10\\"></rect>",
  "<circle cx=\\"10\\" cy=\\"50\\" r=\\"8\\"></circle>",
  "<rect x=\\"25\\" y=\\"45\\" rx=\\"5\\" ry=\\"5\\" width=\\"220\\" height=\\"10\\"></rect>",
  "<circle cx=\\"10\\" cy=\\"80\\" r=\\"8\\"></circle>",
  "<rect x=\\"25\\" y=\\"75\\" rx=\\"5\\" ry=\\"5\\" width=\\"220\\" height=\\"10\\"></rect>",
  "<circle cx=\\"10\\" cy=\\"110\\" r=\\"8\\"></circle>",
  "<rect x=\\"25\\" y=\\"105\\" rx=\\"5\\" ry=\\"5\\" width=\\"220\\" height=\\"10\\"></rect>",
]
`;

exports[`CodeLoader renders its shapes 1`] = `
Array [
  "<rect x=\\"0\\" y=\\"0\\" rx=\\"3\\" ry=\\"3\\" width=\\"70\\" height=\\"10\\"></rect>",
  "<rect x=\\"80\\" y=\\"0\\" rx=\\"3\\" ry=\\"3\\" width=\\"100\\" height=\\"10\\"></rect>",
  "<rect x=\\"190\\" y=\\"0\\" rx=\\"3\\" ry=\\"3\\" width=\\"10\\" height=\\"10\\"></rect>",
  "<rect x=\\"15\\" y=\\"20\\" rx=\\"3\\" ry=\\"3\\" width=\\"130\\" height=\\"10\\"></rect>",
  "<rect x=\\"155\\" y=\\"20\\" rx=\\"3\\" ry=\\"3\\" width=\\"130\\" height=\\"10\\"></rect>",
  "<rect x=\\"15\\" y=\\"40\\" rx=\\"3\\" ry=\\"3\\" width=\\"90\\" height=\\"10\\"></rect>",
  "<rect x=\\"115\\" y=\\"40\\" rx=\\"3\\" ry=\\"3\\" width=\\"60\\" height=\\"10\\"></rect>",
  "<rect x=\\"185\\" y=\\"40\\" rx=\\"3\\" ry=\\"3\\" width=\\"60\\" height=\\"10\\"></rect>",
  "<rect x=\\"0\\" y=\\"60\\" rx=\\"3\\" ry=\\"3\\" width=\\"30\\" height=\\"10\\"></rect>",
]
`;

exports[`FacebookLoader renders its shapes 1`] = `
Array [
  "<rect x=\\"70\\" y=\\"15\\" rx=\\"4\\" ry=\\"4\\" width=\\"117\\" height=\\"6.4\\"></rect>",
  "<rect x=\\"70\\" y=\\"35\\" rx=\\"3\\" ry=\\"3\\" width=\\"85\\" height=\\"6.4\\"></rect>",
  "<rect x=\\"0\\" y=\\"80\\" rx=\\"3\\" ry=\\"3\\" width=\\"350\\" height=\\"6.4\\"></rect>",
  "<rect x=\\"0\\" y=\\"100\\" rx=\\"3\\" ry=\\"3\\" width=\\"380\\" height=\\"6.4\\"></rect>",
  "<rect x=\\"0\\" y=\\"120\\" rx=\\"3\\" ry=\\"3\\" width=\\"201\\" height=\\"6.4\\"></rect>",
  "<circle cx=\\"30\\" cy=\\"30\\" r=\\"30\\"></circle>",
]
`;

exports[`InstagramLoader renders its shapes 1`] = `
Array [
  "<circle cx=\\"30\\" cy=\\"30\\" r=\\"30\\"></circle>",
  "<rect x=\\"75\\" y=\\"13\\" rx=\\"4\\" ry=\\"4\\" width=\\"100\\" height=\\"13\\"></rect>",
  "<rect x=\\"75\\" y=\\"37\\" rx=\\"4\\" ry=\\"4\\" width=\\"50\\" height=\\"8\\"></rect>",
  "<rect x=\\"0\\" y=\\"70\\" rx=\\"5\\" ry=\\"5\\" width=\\"400\\" height=\\"400\\"></rect>",
]
`;

exports[`ListLoader renders its shapes 1`] = `
Array [
  "<rect x=\\"0\\" y=\\"0\\" rx=\\"3\\" ry=\\"3\\" width=\\"250\\" height=\\"10\\"></rect>",
  "<rect x=\\"20\\" y=\\"20\\" rx=\\"3\\" ry=\\"3\\" width=\\"220\\" height=\\"10\\"></rect>",
  "<rect x=\\"20\\" y=\\"40\\" rx=\\"3\\" ry=\\"3\\" width=\\"170\\" height=\\"10\\"></rect>",
  "<rect x=\\"0\\" y=\\"60\\" rx=\\"3\\" ry=\\"3\\" width=\\"250\\" height=\\"10\\"></rect>",
  "<rect x=\\"20\\" y=\\"80\\" rx=\\"3\\" ry=\\"3\\" width=\\"200\\" height=\\"10\\"></rect>",
  "<rect x=\\"20\\" y=\\"100\\" rx=\\"3\\" ry=\\"3\\" width=\\"80\\" height=\\"10\\"></rect>",
]
`;
